### PR TITLE
WIP: add direct-coil free-boundary ESSOS provider lane

### PR DESCRIPTION
**Pull request by rogeriojorge**
Research branch for direct differentiable coil external-field providers in vmec_jax.\n\nCurrent scope:\n- pure JAX provider infrastructure and ESSOS adapter\n- differentiable mgrid interpolation scaffold\n- free-boundary runtime hook for direct coil providers\n- ESSOS Landreman-Paul QA mgrid/direct beta scan example\n- optional ESSOS/VMEC2000 parity gates, with the VMEC2000 generated-mgrid comparison currently marked xfail pending bounded NESTOR parity\n\nLocal tests run:\n- PYTHONPATH=<private-path> pytest -q tests/test_external_fields_coils_jax.py tests/test_external_fields_essos_adapter.py tests/test_external_fields_mgrid_jax.py tests/test_free_boundary_vacuum_adjoint.py tests/test_free_boundary_coil_provider_forward.py tests/test_free_boundary_essos_coil_parity.py::test_essos_direct_coil_free_boundary_matches_generated_mgrid_backend\n- pytest -q tests/test_free_boundary_essos_coil_parity.py\n\nDraft until the full coil-only QS optimization and VMEC2000 generated-mgrid parity lane are completed.

### CODE REVIEW

**Comment by rogeriojorge** [comment]
## Phase-1 validation figures

These figures are hosted outside the repository so the PR does not add binary/figure weight. They summarize the phase-1 completion claim: differentiable direct-coil provider, active finite-pressure free-boundary forward solves, compatibility with generated `mgrid`, validated local gradients for the coil provider, and final CPU/GPU benchmark evidence.

**Scope statement.** Phase 1 is the forward direct-coil free-boundary lane. The full nonlinear free-boundary exact adjoint remains the next scientific phase and is intentionally not claimed here.

### Completion dashboard
Phase-1 completion dashboard

### Direct-coil vs generated-mgrid provider parity
Direct-coil vs mgrid provider parity

### Active finite-pressure free-boundary scan
Finite-pressure free-boundary scan

### Direct-coil gradient validation
Direct-coil gradient validation

### Final CPU/GPU performance evidence
Final CPU/GPU performance evidence

Provenance:
- Branch head: `3758a4e`.
- CI: all required checks green; Physics Full skipped by design for PRs.
- Local full fast gate: `2437 passed, 26 skipped, 112 deselected, 2 xfailed`, coverage `95.00%`.
- Benchmark JSONs used locally: `results/free_boundary_essos_coils_beta_scan_readme/summary.json` and ignored `outputs/bench_freeb_direct_coil_matrix_final_cpu_gpu/summary.json` copied back from office.
- Figure host: https://gist.github.com/rogeriojorge/a49e7a21330fbc8ab99229d2b05de708

**Comment by rogeriojorge** [comment]
## Additional phase-1 WOUT plot comparisons

Added the requested `vmec_jax --plot`-style WOUT comparison across the finite-pressure free-boundary beta scan. This is also hosted outside the repository to avoid adding figure weight.

Phase-1 beta WOUT plot comparison

What it shows:
- iota profiles from the direct-coil free-boundary WOUTs;
- LCFS `|B|` contour-line plots for each beta point;
- LCFS cross-section overlays at two toroidal cuts;
- magnified LCFS displacement and scalar shape-response metrics relative to beta=0.

Important interpretation: this bounded phase-1 scan is intentionally low pressure. The WOUT-derived beta proxy only reaches `1.38e-4%`, so the Shafranov/shape response is real but very small: max LCFS displacement is about `1.1e-7`, mean axis radial shift about `2.6e-8`, and max LCFS `|B|` change about `7.3e-7`. The figure therefore shows both physical overlays and a magnified/quantified displacement panel. A stronger finite-beta production scan should be a follow-up validation case, not a hidden assumption in this phase-1 PR.

Provenance: generated from `results/free_boundary_essos_coils_beta_scan_readme/wout_direct_beta_*.nc` using the same WOUT Fourier data/evaluation path behind `vmec_jax --plot`.

**Comment by rogeriojorge** [comment]
Added the actual-WOUT-beta finite-pressure validation requested before review.

Key points:

- The previous phase-1 beta scan used the nominal input label; the WOUT value was only `100*wp/wb = 1.38e-4%`.
- The new direct-coil validation calibrates the VMEC flux scale to the ESSOS coil field with `PHIEDGE=-3.0e-2` and reports the actual WOUT beta from `100*wp/wb`.
- The final direct-coil WOUT case reaches `100*wp/wb = 0.9834%`, `aspect = 5.99999`, `<iota> = 0.32894`, and final `FSQ total = 4.87e-2`.
- The figure uses WOUT data analogous to `vmec_jax --plot`: iota profiles, LCFS `|B|` contour-line plots, cross-sections, and amplified LCFS displacement/field-response diagnostics across beta.
- I pushed `aaa8764` adding reproducible example knobs: `--phiedge` and `--pressure-scale-for-one-percent-beta`. Reproduction command is now documented in `docs/free_boundary_coil_optimization.rst`.

Actual WOUT beta direct-coil free-boundary validation

Raw summary JSON: https://gist.githubusercontent.com/rogeriojorge/a49e7a21330fbc8ab99229d2b05de708/raw/643d33c97d4aa46f21a21c84867b524b06f04d7b/phase1_actual_beta_phi03_summary.json

**Comment by rogeriojorge** [comment]
Reopened after force-slim main refresh; branch has been rebuilt on current main without reintroducing trimmed WOUT/figure artifacts.
